CRM/Core/Smarty/plugins/block.edit.php

```diff
@@ -38,7 +38,8 @@
  */
 function smarty_block_edit($params, $text, &$smarty, &$repeat) {
   if (!$repeat) {
-    $action = $smarty->getTemplateVars()['action'];
+    $b = $smarty->getTemplateVars();
```

Comment:
I would use $v (grin).

I'll assume you're confident this works in all versions but I was also going to check universe if extensions are doing this. If it's a lot then maybe worth having some BC code, or consider posting an upstream revert given v5 is 2 weeks old. I don't know why smarty has done this breakage.

Reply:
b is way more correct

Reply:
It's always been an option to pass the variable name in - ie here is the Smarty2 function

![image](https://github.com/civicrm/civicrm-core/assets/336308/0ba2a62b-a209-4da4-9933-6e07ab9a8c1d)

Reply:
@demeritcowboy this only place in Universe I could find that would need to make this change is advancedevents managed by @mattwire 

![image](https://github.com/civicrm/civicrm-core/assets/336308/1706d8dd-cc31-41f8-9a06-0b42268a8b68)


would need to change to 



    $rows = $page->getTemplateVars('rows');


Note that is going through `get_template_vars()`  - it might make sense to type-hint that function to give a hard message that would probably be less confusing 


![image](https://github.com/civicrm/civicrm-core/assets/336308/a699d1b0-7df4-48f7-b4ea-6e3c53c3a0fd)